### PR TITLE
feat: support for filecoin native addresses

### LINKED ISSUE #10433: Support Filecoin native addresses representation in API/UI
Converting EVM address to Filecoin address requires making additional calls to Lotus node endpoints:

> This just requires calling Filecoin.EthAddressToFilecoinAddress to get the Filecoin short ID (which depends on current state) and then Filecoin.StateLookupRobustAddress to get the final public-key address ([Lotus Docs](https://github.com/filecoin-project/lotus/blob/61f29a84b5a61060c4ac8aabe9b9360a2cdf5e7e/documentation/en/api-v1-unstable-methods.md) for all these methods).

Requirements to the implementation:
- We should extend `addresses` table with a column to store Filecoin representation of addresses.
- Regular block import should fill this column.
- We need a backfiller of existing addresses: it should process existing addresses in batches with ability to setup batch size and concurrency via env variables and committing its status into `migration_status` table.
- Pass Filecoin address representation to the corresponding API v2 endpoints together with EVM address in order to reflect the changes in UI.
- All above is applied to :filecoin chain type only.

### PR DESCRIPTION
**Pull request by fedor-ivn**
Closes #10433.

### Enhancements

This PR introduces support for Filecoin native addresses in Blockscout.
#### Fetching Native Address Information

The process for fetching Filecoin native address information is similar to the existing mechanism for the internal transactions fetcher. Fetching Filecoin address information is resource-intensive, as the Beryx API — an external provider of indexed Filecoin data — does not support batch requests. Consequently, addresses are fetched one by one. The Beryx API enables Blockscout to map Ethereum addresses to Filecoin counterparts.

During the indexing process, any imported address will generate an entry in the `filecoin_pending_address_operations` table. The `Indexer.Fetcher.Filecoin.AddressInfo` fetcher later processes these entries, retrieves the necessary Filecoin address details, updates the corresponding fields in the `addresses` table , and removes the pending operation.
#### Storing Filecoin addresses in DB

Filecoin addresses are stored as raw bytes in newly added fields within the `addresses` table: `filecoin_id`, `filecoin_robust`, and `filecoin_actor_type`. For the encoding of these addresses, a custom Ecto-type was introduced. For more details, refer to the documentation for the `Explorer.Chain.Filecoin.NativeAddress` module.

#### API updates

API responses that include address objects are now extended with a `filecoin` section containing information about Filecoin native addresses. Here is an example of the response returned by `/api/v2/addresses/{address_hash}`:
```jsonc
{
	"filecoin": {
		"id": null,
		"robust": null,
		"actor_type": null
	},
	"block_number_balance_updated_at": 4160473,
	"coin_balance": "158353053030223974150924",
	"creation_tx_hash": null,
	"creator_address_hash": null,
	"ens_domain_name": null,
	"exchange_rate": null,
	"has_beacon_chain_withdrawals": false,
	"has_decompiled_code": false,
	"has_logs": false,
	"has_token_transfers": false,
	"has_tokens": false,
	"has_validated_blocks": false,
	"hash": "0xFf000000000000000000000000000000002bb4AC",
	"implementations": [],
	"is_contract": false,
	"is_verified": false,
	"metadata": null,
	"name": null,
	"private_tags": [],
	"public_tags": [],
	"token": null,
	"watchlist_address_id": null,
	"watchlist_names": []
}
```
#### New Background Migration

This PR introduces the filling migration `Explorer.Migrator.FilecoinPendingAddressOperations`, which populates the `filecoin_pending_address_operations` table with historical addresses that were imported into the database before the changes introduced in this PR and do not yet have corresponding Filecoin native address information. 

#### New Metric

This PR adds a new private gauge metric, `filecoin_pending_address_operations`. This metric tracks the number of records in the `filecoin_pending_address_operations` table, indicating how many addresses are still awaiting their native address information to be fetched.

#### New ENVS

Details on the new environment variables can be found in the related [PR to docs repo](https://github.com/blockscout/docs/pull/313).

### Upgrading

To properly configure Filecoin native addressing, the instance maintainer must obtain a Beryx API token, which can be acquired [here](https://docs.zondax.ch/beryx-api), and set it in the `BERYX_API_TOKEN` environment variable.

## Checklist for your Pull Request (PR)

  - [x] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [x] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [x] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
https://github.com/blockscout/docs/pull/313
  - [ ] If I added new DB indices, I checked, that they are not redundant with PGHero or other tools.
  - [ ] If I added/removed chain type, I modified the Github CI matrix and PR labels accordingly.

### CODE REVIEW

File: apps/explorer/lib/explorer/chain/import/runner/helper.ex [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  @doc """
+  Executes the import function if the configured chain type matches the
+  specified `chain_type`.
+  """
+  @spec chain_type_dependent_import(
+          Ecto.Multi.t(),
+          chain_type :: atom(),
+          (Ecto.Multi.t() -> Ecto.Multi.t())
+        ) :: Ecto.Multi.t()
+  def chain_type_dependent_import(multi, chain_type, multi_run) do
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/chain/import/runner/helper.ex:15]: Can we use this function in `Explorer.Chain.Import.Runner.Blocks` module as well?

File: apps/explorer/lib/explorer/chain/filecoin/native_address.ex [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+
+  @min_address_string_length 3
+
+  # Payload sizes:
+  # f1 -- 65 bytes
+  # f2 -- 32 bytes
+  # f3 -- 48 bytes
+  @protocol_indicator_to_payload_byte_count %{
+    1 => 20,
+    # todo: WTF? Should be 32. Specs are lying
```
Comment (vbaranov) [inline apps/explorer/lib/explorer/chain/filecoin/native_address.ex:62]: 🤔

Comment (fedor-ivn) [reply apps/explorer/lib/explorer/chain/filecoin/native_address.ex:62]: forgot to remove :see_no_evil:

File: apps/indexer/lib/indexer/fetcher/filecoin/address_info.ex [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+  @spec run([Explorer.Chain.Filecoin.PendingAddressOperation.t(), ...], any()) :: :ok | :retry
+  def run([pending_operation], _state) do
+    fetch_and_update(pending_operation)
+  end
+
+  @spec fetch_and_update(PendingAddressOperation.t()) :: :ok | :retry
+  defp fetch_and_update(%PendingAddressOperation{address_hash: address_hash} = operation) do
+    with {:ok, new_params} <- fetch_address_info_using_beryx_api(operation),
+         {:ok, _} <- update_address_and_remove_pending_operation(operation, new_params) do
+      Logger.info("Fetched Filecoin address info for: #{to_string(address_hash)}")
```
Comment (fedor-ivn) [inline apps/indexer/lib/indexer/fetcher/filecoin/address_info.ex:106]: This log might appear too often. Should we consider reducing log level to `debug`?

File: apps/explorer/priv/filecoin/migrations/20240820080027_add_http_status_code.exs
[the remark this answers is not in the document]
Comment (fedor-ivn) [reply apps/explorer/priv/filecoin/migrations/20240820080027_add_http_status_code.exs:1]: Done :heavy_check_mark:

File: apps/explorer/priv/filecoin/migrations/20240807134138_add_chain_type_fields_to_address.exs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    alter table(:addresses) do
+      add(:filecoin_id, :bytea)
+      add(:filecoin_robust, :bytea)
+    end
+
+    execute(
+      "ALTER TABLE addresses ADD COLUMN filecoin_actor_type SMALLINT;",
+      "ALTER TABLE addresses DROP COLUMN filecoin_actor_type;"
+    )
+  end
```
[the remark this answers is not in the document]
Comment (fedor-ivn) [reply apps/explorer/priv/filecoin/migrations/20240807134138_add_chain_type_fields_to_address.exs]: Good catch! Done :heavy_check_mark:

File: apps/indexer/lib/indexer/fetcher/filecoin/address_info.ex (hunk 3 above)
Comment (vbaranov) [reply apps/indexer/lib/indexer/fetcher/filecoin/address_info.ex:106]: Yes, please, let's reduce to `debug`.

**Review by vbaranov — APPROVED** [review]

File: apps/indexer/lib/indexer/fetcher/filecoin/address_info.ex (hunk 3 above)
Comment (fedor-ivn) [reply apps/indexer/lib/indexer/fetcher/filecoin/address_info.ex:106]: :heavy_check_mark:

File: apps/indexer/lib/indexer/prometheus/collector/filecoin_pending_address_operations_collector.ex [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  alias Explorer.Chain.Filecoin.PendingAddressOperation
+  alias Explorer.Repo
+  alias Prometheus.Model
+
+  def collect_mf(_registry, callback) do
+    callback.(
+      create_gauge(
+        :filecoin_pending_address_operations,
+        "Number of records in filecoin_pending_address_operations table",
+        Repo.aggregate(PendingAddressOperation, :count)
```
[the remark this answers is not in the document]
Comment (fedor-ivn) [reply apps/indexer/lib/indexer/prometheus/collector/filecoin_pending_address_operations_collector.ex]: Good catch, processed in https://github.com/blockscout/blockscout/pull/10468/commits/c63043e290a6b0612272e1c03a64447dfc6cd818

File: apps/explorer/lib/explorer/migrator/filecoin_pending_address_operations.ex [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        ordered_pending_operations,
+        conflict_target: :address_hash,
+        on_conflict: :nothing,
+        for: PendingAddressOperation,
+        returning: true,
+        timeout: :infinity,
+        timestamps: Import.timestamps()
+      )
+    end)
+    |> Repo.transaction()
```
[the remark this answers is not in the document]
Comment (fedor-ivn) [reply apps/explorer/lib/explorer/migrator/filecoin_pending_address_operations.ex]: Good catch. I don't have an idea why I used Multi here. Processed your comment in https://github.com/blockscout/blockscout/pull/10468/commits/9f3155e002a2dea6e501a939ff20b74d16fa8d43 and leaved a call to `Import.insert_changes_list` only.

**Pull request merged**